src/StudentList.java

```diff
@@ -0,0 +1,37 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+//Studentクラスのコンストラクタを使用し、リストを作成。
+public class StudentList {
+    public static void main(String[] args) {
+
+        List<Student> list = List.of(
+                new Student("suzuki",180,"A"),
+                new Student("konnno",170,"A"),
+                new Student("saitou",158,"B"),
+                new Student("tajima",165,"O"),
+                new Student("yamaguchi",148,"B"),
+                new Student("suzuki",153,"O"));
+
+        //nameの文字列に"s"を含む要素を取り出す
+        List<Student> nameList = list.stream().filter(n->n.getName().contains("s")).collect(Collectors.toList());
+        System.out.println(nameList);
+
+        //heightが160以上の値を持つ要素を取り出す
+        List<Student> heightList = list.stream().filter(n->n.getHeight() >= 160).collect(Collectors.toList());
+        System.out.println(heightList);
+
+        //血液型でグルーピング
+        Map<String,List<Student>> grp = list.stream()
```

Comment:
Mapの変数名はわかりにくくなりがちなのですが、個人的には
keyToValue
か
valueByKey
のような形式が良いと思っています。
```suggestion
        Map<String, List<Student>> bloodTypeToStudents = list.stream()
```